qubesfiletrust/qvm_file_trust.py

```diff
@@ -0,0 +1,546 @@
+    try:
+        with open(path):
+            pass
+
+    except IOError:
+        # Try to unlock file to get read access
+        safe_chmod(path, 0o644,
+            'Could not unlock {} for reading'.format(path))
+    if trusted:
+        # Set file to trusted
+        # AKA remove our xattr
+        file_xattrs = xattr.get_all(path)
+        untrusted_attribute = (b'user.qubes.untrusted', b'true')
+
+        # Check if the xattr exists first
+        if untrusted_attribute in file_xattrs:
+            try:
+                xattr.removexattr(path, 'user.qubes.untrusted')
+            except:
+                # Unable to remove our xattr, return original permissions
+                error('Unable to remove untrusted attribute on {}'.format(path))
+                safe_chmod(path, orig_perms,
+                    'Unable to set original perms. on {}'.format(path))
+
+        # Remove visual attributes
+        set_visual_attributes(path, False)
+
+    else:
+        # Set file to untrusted
+        # AKA add our xattr and lock
+        try:
+            xattr.setxattr(path, 'user.qubes.untrusted', 'true')
+            safe_chmod(path, 0o0,
+                    'Unable to set untrusted permissions on: {}'.format(path))
+        except:
+            # Unable to remove our xattr, return original permissions
+            safe_chmod(path, orig_perms,
+                'Unable to return perms after setting as untrusted: {}'.
+                format(path))
+            sys.exit(65)
+
+        # Add visual attributes
+        set_visual_attributes(path, True)
+
+def change_folder(path, trusted):
+    """Change the trust state of a folder"""
+
+    # Remove '/' from end of path
+    path = os.path.normpath(path)
+
+    try:
+        # Create the ~/.config/qubes folder if it doesn't exist
+        home_dir = os.path.expanduser('~')
+        if not os.path.exists(home_dir + '/.config'):
+            os.mkdir(home_dir + '/.config')
+        if not os.path.exists(home_dir + '/.config/qubes'):
+            os.mkdir(home_dir + '/.config/qubes')
+
+        # Create the local file if it does not exist
+        if not os.path.exists(LOCAL_FOLDER_LOC):
+            open(LOCAL_FOLDER_LOC, 'a').close()
+    except:
+        error('Could not create local rule list: {}'.format(
+                LOCAL_FOLDER_LOC))
+        error('Check /home/<your user> folder exists...')
+        sys.exit(72)
+
+    if trusted:
+        # Set folder to trusted
+        # AKA remove any mentions from untrusted paths list
+        try:
+            file = open(LOCAL_FOLDER_LOC, 'r+')
+        except:
+            error('Unable to read local untrusted folder: {}'.
+                    format(LOCAL_FOLDER_LOC))
+            sys.exit(72)
+
+        lines = file.readlines()
+        file.seek(0)
+
+        # Write back all lines to the file except ones containing our path
+        for line in lines:
+            if line.rstrip() != path:
+                file.write(line)
```

Comment:
If it wasn't there, this function should at least report a warning. But probably adding an exclusion (`-` prefix) is a good idea - if the path is in the global list.

Reply:
Good idea on the exclusion prefix. I cleaned up a bit of code in that area to also delete existing explicitly-trusted rules in the local list if we were setting as untrusted again. 

Also added the warning. 45441f2